### PR TITLE
✨ Implement Data Mapper GET Consent Wizard Progress (#296) (PR 2)

### PR DESCRIPTION
**Pull request by mistryrn**
## Description of Changes

PR 2 of 3.

Implements the `GET /wizard/progress/:participantId` endpoint in Data Mapper, adds supporting types. Also fixes a small bug in `getConsentParticipantById` in Data Mapper.

### Data Mapper
* Adds `GET /wizard/progress/:participantId` router to Data Mapper
* Fixes small bug in Data Mapper `getConsentParticipantById` search service function (return parsed `ConsentParticipantResponse` instead of raw data which could have `null` instead of `undefined`)

### Types
* Add `GetConsentProgressResponse` response type to `types/dataMapper`

### Special Instructions
Before running these changes, you should rebuild the `types` package:
```bash
pnpm build:types
```

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [ ] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [ ] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

File: apps/data-mapper/src/services/das/consent.ts [hunk 1]
```diff
@@ -323,7 +323,7 @@ export const getConsentParticipantById = async (
 			logger.error('Received invalid data fetching participant', participant.error.issues);
 			return failure('SYSTEM_ERROR', participant.error.message);
 		}
-		return success(result.data);
+		return success(participant.data);
```
Comment (mistryrn) [inline apps/data-mapper/src/services/das/consent.ts:326]: Smol bug I noticed while working on this – we were parsing/validating the participant but not returning parsed data.

File: packages/types/src/entities/ConsentWizardProgress.ts [hunk 2]
```diff
@@ -17,6 +17,9 @@
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+// TODO: this consent wizard progress structure has been deprecated and should be removed
+// https://github.com/OHCRN/platform/issues/661
```
Comment (mistryrn) [inline packages/types/src/entities/ConsentWizardProgress.ts:21]: Adding a note here to remove this old Consent Wizard Progress type as part of our cleanup tasks.

File: packages/types/src/services/dataMapper/responses/ConsentParticipant.ts [hunk 3]
```diff
@@ -21,13 +21,15 @@ import { z } from 'zod';
...
 import { ConsentParticipant, LifecycleState } from '../../../entities/index.js';
+import { ConsentCategory } from '../../../entities/fields/index.js';
 
 export const ConsentParticipantResponse = ConsentParticipant.extend({
 	previousLifecycleState: LifecycleState.nullable().transform((input) => input ?? undefined),
 	assentFormAcknowledged: z
 		.literal(true)
 		.nullable()
 		.transform((input) => input ?? undefined),
+	consentProgress: ConsentCategory.nullable().transform((input) => input ?? undefined),
```
Comment (mistryrn) [inline packages/types/src/services/dataMapper/responses/ConsentParticipant.ts:32]: There are a couple other ConsentParticipant-related response types that need this transformation, I fixed those in PR 3 of 3: https://github.com/OHCRN/platform/pull/734

File: apps/consent-das/prisma/schema.prisma [hunk 4]
```diff
@@ -43,6 +43,7 @@ model Participant {
   ParticipantResponse    ParticipantResponse[]
   Signature              Signature[]
   assentFormAcknowledged Boolean?
+  consentProgress        ConsentCategory?
```
Comment (anncatton) [inline apps/consent-das/prisma/schema.prisma:46]: any idea why git isn't recognizing the commits from the base branch for the migration changes?

**Review by anncatton — APPROVED** [review]
this looks good, just wondering why the commits from the base branch are being shown as changes in this one 🤔

File: apps/consent-das/prisma/schema.prisma (hunk 4 above)
Comment (mistryrn) [reply apps/consent-das/prisma/schema.prisma:46]: Probably just need to rebase this with the latest base branch – will do that after PR 3 gets merged into this one, before requesting re-review/approval to merge into the base :)

**Pull request merged**